File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    /*
     [0,1,0,3,12]
     131200
    
    
    */
    void moveZeroes(vector<int>& nums) {
      int n=nums.size();
        vector<int>ans;
        int i,count=0;
        for(i=0;i<n; i++){
            if(nums[i]==0){//when encounter a 0 we increament count
                count++;
            }else{//add non zero elements to ans vector
                ans.push_back(nums[i]);
            }
        }
        for(i=0; i<count; i++){//we push back count number of 0's to the end
            ans.push_back(0);
        }
        for(i=0;i<n; i++){//put all the elements on ans vector to nums
           nums[i]=ans[i];
        }
        //return ans;
    }
      /* void moveZeroes(vector<int>& nums) {
        int n=nums.size(),r=0,l=0;
           if(n==0||n==1) return;
           while(r<n){
               if(nums[r]==0){//if 0 move r ptr
                   r++;
               }else{//if not swap with the element at the l index and increament both r&l ptr
                   swap(nums[l++], nums[r++]);
               }
           }
    }*/
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        length = len(nums)
        right = 0
        left =0 
        while(right < length):
            if nums[right] == 0:
                right += 1;
            else:
                temp = nums[right];
                nums[right] = nums[left]
                nums[left] = temp
                right += 1
                left += 1
                